ee/localserver/zta_auth_middleware.go

```diff
@@ -0,0 +1,285 @@
+package localserver
+
+import (
+	"crypto/ecdsa"
+	"encoding/base64"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"log/slog"
+	"net/http"
+	"net/url"
+	"time"
+
+	"github.com/kolide/krypto/pkg/echelper"
+	"github.com/lestrrat-go/jwx/jwk"
+)
+
+type ztaAuthMiddleware struct {
+	counterPartyKeys map[string]*ecdsa.PublicKey
+	slogger          *slog.Logger
+}
+
+type ztaResponse struct {
+	Data   []byte    `json:"data"`
+	PubKey *[32]byte `json:"pubKey"`
+}
+
+func (z *ztaAuthMiddleware) Wrap(next http.Handler) http.Handler {
+	return http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		boxParam := r.URL.Query().Get("payload")
+		if boxParam == "" {
+			z.slogger.Log(r.Context(), slog.LevelDebug,
```

Comment:
Do we potentially want to log these errors at a higher level than debug, so we ship them to the cloud? `LevelWarn` or `LevelError` both seem appropriate to me